src/aiida/tools/pytest_fixtures/storage.py

```diff
@@ -75,6 +78,10 @@ def config_psql_dos(
     def factory(
         database_name: str | None = None, database_username: str | None = None, database_password: str | None = None
     ) -> dict[str, t.Any]:
+        if not postgres_cluster:
+            msg = 'Could not initialize PostgreSQL cluster'
```

Comment:
Good point. I guess I was trying to reduce the depth of stacktrace that pytest vomits when this happens. But it could hide what is the actual reason for the error. I've changed it.

Reply:
I take that back --- this was a load-bearing hack, because this fixture is evaluated eagerly and before we could decide we want `sqlite_dos` backend in `aiida_profile`. I'd need to think about a nicer solution than this since this.

Reply:
> a load-bearing hack

🤣 I love it
That makes sense that you have to fix it this way, but I wonder what will happen if someone uses the `postgres_cluster` fixture. They will just get `None` instead of an exception which won't be immediately obvious. Please just comment the reason for the `OSError` and add a line in the docstring when it will return `None`.

Reply:
Yeah, since this is a public fixture the hack is really not ideal. Since this is a new fixture that hasn't been released yet, should we try to change its API so that the cluster is created only when really needed?

Reply:
If you have an idea of how, then yes I think that would be preferable

Reply:
I tried to refactor by encapsulating the cluster in a class / object. LMK if it makes sense. 
If the only thing that users of this fixtures are supposed to interact with is the `create_database` method, then I think this new approach makes sense.